### ISSUE TITLE
Create compelling examples to demonstrate Timeplus Proton and publish videos 

### ISSUE THREAD

**Issue by jovezhong** [timeplus-io/proton#527]
## Description

In the Proton [examples](https://github.com/timeplus-io/proton/tree/develop/examples) folder, we have created a few examples to demonstrate Proton's fast/lightweight/powerful stream processing capabilities, such as the [cdc](https://github.com/timeplus-io/proton/tree/develop/examples/cdc) example to load MySQL changes via Debezium and Redpanda, the [carsharing](https://github.com/timeplus-io/proton/tree/develop/examples/carsharing) example to showcase how to push data to Proton without Kafka API, the [awesome-sensor-log](https://github.com/timeplus-io/proton/tree/develop/examples/awesome-sensor-logger) to demonstrate how to visualize the sensors on your phone via Proton and Grafana, etc.

We need more examples to attract developers in different industries to explore how to use Proton to simplify their streaming ETL, or real-time analytics. 

**Required**:
Please create a new PR with easy setup. It can be a docker-compose with a set of preconfigured data generator, or real-world data sources, with sample SQL or even dashboards/charts(with Grafana, or Metabase, or Streamlit). In the PR , we also expect a README to explain the use case, key Proton features to demo. Feel free to publish this README somewhere else as a blog.

**Nice to have**
* Publish the README somewhere else as a blog
* A video to demo how Proton makes stream processing easier or solve the issues.

## Acceptance Criteria

Timeplus team will review the entire PR, ensuring the README is well documented, code is clean and easy to understand. If the contributor makes a video, it should well explain how the demo app works and highlights the key Proton features or strengths. 

As the community suggests, we will reward for both **effort** and **reach**. Of course, with the focus of quality. 

### Effort

|Basic|Award|
|--|--|
| a meaningful PR created|$25|

Anyone who creates a meaningful PR will get $25 to appreciate your effort and contribution. Sorry for the vague term "meaningful". A simple hello world is not meaningful. Copying an existing demo or making very minor changes are not expected either. The Timeplus team will reserve the rights to judge whether the PR is meaningful or not. As long as you put effort trying to build a good demo for Timeplus Proton, we will reward you, even the PR is not merged. We will send you the $25 reward or tip 7 days since you create the PR.

The README in the new Proton example can be published as a blog in elsewhere. When we merge your PR to Proton `develop` branch, we will also send you a reward based on the content of the README/blog. Assuming the quality of the README (we may send you PR comments before merging it), the reward is based on the effort.

|README/blog word count|Award|
|--|--|
|PR merged and the word count is 500-1000|$75|
|PR merged and the word count is 1001-1500|$150|
|PR merged and the word count is 1500 or above|$200|

 ### Reach

To measure the impact of the video, we need the video to be published on YouTube. Only the first 30 days' view count is considered and will be rewarded:

|Demo video views count (30 days)|Award|
|--|--|
|video view count is below 100|$50|
|video view count is 101-1000|$200|
|video view count is 1001-2000|$300|
|video view count is over 2001|$400|

### Number of attempts or number of rewards
Until we close this ticket(which probably won't happen in 2024), we are expecting multiple demo/blog/video to be contributed to Proton. One person can try multiple times. While we hope we keep this ticket open for long term and keep rewarding our contributors, Timeplus reserves the rights to close it any time.

### Still confused for the amount and timeline?
Here is an example.

|Date|Activity|Award|
|--|--|--|
|April 2, 2024|DevA creates a PR and marks for review||
|April 5, 2024|Timeplus team confirms it's a meaningful PR|$25|
|April 10, 2024|Timeplus team merged the PR with 800 words in the README|$75|
|May 1, 2024|video gets 800 views for fist 30 days|$200|
||Total|$300|

**Comment by Akshay-SL** [comment]
/attempt #527

<div id="algora" />

<details>
  <summary>Options</summary>
  <ul>
    <li>
      <a href="https://console.algora.io/api/bounties/clrwwoohx0004jw0gkls6nw5o/cancel-attempt">
        Cancel my attempt
      </a>
    </li>
  </ul>
</details>

**Comment by jovezhong** [comment]
Thank you @Akshay-SL to pick up this task. Feel free to join https://timeplus.com/slack if you want to discuss live with us. Leaving comments in this ticket or sending PR also work

**Comment by ayewo** [comment]
Hi @jovezhong 👋,

I'd like to take a stab at this but the task as specified suffers from a mismatch between the effort required and the incentives offered.

To put it bluntly, the suggested payment on this issue is unrealistic.

## Two Goals
It seems there are 2 goals you are trying to nail with this:
a. produce sample code to go with a compelling article;
b. produce a demo video to accompany the article.

For a., there's no indication of article length. The most common way to approximate the effort it takes to produce a technical article is to use the word count.

With respect to b., you are expecting to generate some inbound interest in T+ via the demo videos you are hoping the community will create. You are willing to reward community members that generate **marketing exposure** for T+ via their demo videos posted to YouTube. This is why the reward is tied to the 30-day view count for each video.

## Effort
The **effort** for researching and authoring a compelling article is definitely worth more than $50. Even if we peg the payment for such an article at $50, doing a demo video to go with the article is also worth something but definitely not worth the multiples you are offering to pay. 

The **effort** it takes to author a companion video is certainly _not worth_ x4 times ($250 - $50 = $200) nor is it worth x9 times ($500 - $50 = $450) the **effort** it takes to research and write a compelling article. 

## Reach
You are hoping to reach new audiences via YouTube and we can simply call this **reach**, for short. It's better to separate **reach** from the **effort** required to produce an article and companion video.

## Proposal
In practice, you want to reward two things: **effort** and **reach** but your suggested payment only rewards **reach**.

| Word count 	| Award |
|----------------------------|---------|
| PR merged and the word count is 500-1000 words |	$150 |
| PR merged and the word count is 1000-1500 words | $200 |
| PR merged and the word count is 1500-2000 words | $250 |


| Views count (30 days) 	| Award |
|----------------------------|---------|
| Video count is below 100 |	$50 |
| Video count is 101-1000 |	$150 |
| Video count is 1001-20000 | $250 |

In the current form, once an article's PR is merged, the contributor does not get paid **anything** until 30 days later. 

In this proposal, once an article's PR is merged, the contributor gets paid immediately for the **effort** it took to produce the article and the demo video. The payment for **reach**—marketing exposure attributable to the demo video can come 30 days later when the view count is available from YouTube.

I strongly suggest you restructure the payment to reward **effort** and **reach** individually.

**Comment by jovezhong** [comment]
Thank you [Saïd](https://github.com/ayewo) for the great suggestions. I have updated the description of the GitHub issue, largely based on your feedbacks and suggestions.

We will send $25 reward for any meaningful PR, even it's not merged. Then based on the length of the README/blog, different award to appreciate the effort. The video view count is not something can be controlled. We appreciate the marketing exposure.

The video can be uploaded to our Timeplus YouTube channel but can be in other channels too.

**Comment by ayewo** [comment]
Thanks for accepting my suggestions 🙏 and I’m glad that my reframing here was more effective than my original framing in T+ Slack

You probably want to add a warning somewhere that any submissions that copy-paste verbatim from ChatGPT / Sora will be considered low-effort.

**Comment by jovezhong** [comment]
GenAI in such creative work is always a fun topic. I would say using ChatGPT in the README or Sora in the video is not prohibited. Make sure the content is in high quality. If we feel the content in the PR is too robotic, we will share review comments and it will take longer to get the PR merged.

**Comment by 0xnemian** [comment]
bounty still open? @jovezhong

**Comment by chenziliang** [comment]
Hi @0xmoner, the bounty is not available any more. Thanks for your interest. Would love to check with your demo idea if you like chat. Thanks !

**Comment by jacanidjar** [comment]
@jovezhong  this bounty open to try?

**Comment by jovezhong** [comment]
No, it's not. Please check with @chenziliang for more details
